script.js

```diff
@@ -1,12 +1,363 @@
+    defaultOptionShow.textContent = "Select show...";
+    defaultOptionShow.setAttribute("value", "default");
+    showDropDown.appendChild(defaultOptionShow);
+
+    showsData.forEach((show) => {
+      const listOfShows = document.createElement("option");
+      listOfShows.textContent = `${show.name}`;
+      showDropDown.appendChild(listOfShows);
+    });
+
+    showDropDown.addEventListener("change", async function (event) {
+      if (event.target.value === "default") {
+        searchField.setAttribute("placeholder", "Search for show/s...");
+        searchField.removeEventListener("input", searchFilterEpisodes);
+        searchField.addEventListener("input", searchFilterShows);
+        searchField.value = "";
+        displayingFilterResults.textContent = "";
+        removeCardLayout();
+        clearCards();
+        render();
+        return;
+      } else {
+        searchField.setAttribute("placeholder", "Search for episode/s...");
+        const findID = showsData.find(
+          (show) => show.name === event.target.value
+        );
+        currentState.showsID = findID.id;
+        currentState.allEpisodes = await getAllEpisodes();
+        addCardLayout();
+        clearCards();
+        makePageForEpisodes(currentState.allEpisodes);
+        currentState.episodesFetched = false;
+      }
+    });
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+function dropDownAfterSearch(currentShows) {
+  showDropDown.innerHTML = "";
+  currentShows.forEach((show) => {
+    const option = createElement("option", "");
+    option.textContent = show.name;
+    showDropDown.appendChild(option);
+  });
+}
+
+const searchField = document.querySelector("#search__field");
+
+if (searchField.getAttribute("placeholder") === "Search for show/s...") {
+  searchField.addEventListener("input", searchFilterShows);
+} else if (
+  searchField.getAttribute("placeholder") === "Search for episode/s..."
+) {
+  searchField.addEventListener("input", searchFilterEpisodes);
+}
+
+async function searchFilterShows() {
+  const showsData = await getAllShows();
+  const showsFilter = showsData.filter(
+    (show) =>
+      show.name.toLowerCase().includes(searchField.value.toLowerCase()) ||
+      show.summary.toLowerCase().includes(searchField.value.toLowerCase())
+  );
+
+  if (searchField.value.length > 0) {
+    console.log("filtering shows");
+    displayingFilterResults.textContent = `Found ${showsFilter.length} shows`;
+    removeCardLayout();
+    clearCards();
+    makePageForShows(showsFilter);
+  } else {
+    displayingFilterResults.textContent = "";
+    removeCardLayout();
+    clearCards();
+    render();
+    showDropDown.value = "default";
+  }
+
+  searchField.removeEventListener("input", searchFilterEpisodes);
+  searchField.addEventListener("input", searchFilterShows);
+  const searchLayout = document.querySelector(".search__layout");
+  searchLayout.appendChild(displayingFilterResults);
+}
+
+async function searchFilterEpisodes() {
```

Comment:
I was unable to use the filter episodes feature. When I navigate to the episodes page for a certain show and type something in the search field, the page takes me back to the search results for shows instead.